src/main/kotlin/com/fortysevendeg/arrowinpractice/endpoints/CharacterDetails.kt

```diff
@@ -18,15 +52,36 @@ import io.ktor.routing.get
  */
 fun Routing.characterDetailsEndpoint(charactersDB: CharactersDatabase) {
   authenticate {
-    get("/characters/{characterId}") {
-      val param = call.request.path().substringAfterLast("/")
-      try {
-        val characterId = param.toLong().characterId()
-        val maybeCharacter = charactersDB.getById(characterId)
-        maybeCharacter?.let { character -> call.respond(character) } ?: throw NotFoundException()
-      } catch (e: NumberFormatException) {
-        throw InvalidIdException()
+    getIO("/characters/{characterId}") { app ->
+      monad().binding {
+        val maybeId: Option<String> = paramOf("characterId", app)
+        val characterIdString: String = characterIdOrNotFound(maybeId).bind()
+        val characterId = stringIdToLong(characterIdString).bind()
+        val character = handleDBExceptions(fetchCharacterById(charactersDB, characterId)).bind()
+        character
       }
     }
   }
 }
+
+@JvmName("kindedGet")
```

Comment:
I think we should hide all these utils in a super class / interface or something to make it transparent for them, not on the same file. If the code using those is written from the beginning, they don't even need to care much during the workshop.